src/mscorlib/shared/System/StringSpanHelpers.cs

```diff
@@ -125,5 +125,38 @@ public static ReadOnlySpan<char> Remove(this ReadOnlySpan<char> source, int star
             source.Slice(startIndex + count).CopyTo(result.Slice(startIndex));
             return result;
         }
+
+        // Returns the index of the last occurrence of a specified character in the current instance.
+        // The search starts at startIndex and runs backwards to startIndex - count + 1.
+        // The character at position startIndex is included in the search.  startIndex is the larger
+        // index within the string.
+        //
+        public static int LastIndexOf(this ReadOnlySpan<char> source, char value)
+        {
+            return LastIndexOf(source, value, source.Length - 1, source.Length);
+        }
+
+        public static int LastIndexOf(this ReadOnlySpan<char> source, char value, int startIndex)
+        {
+            return LastIndexOf(source, value, startIndex, startIndex + 1);
+        }
+
+        public static int LastIndexOf(this ReadOnlySpan<char> source, char value, int startIndex, int count)
```

Comment:
Can we just slice the source span instead of having the startIndex and startIndex/count overloads?